Sources/NIOExtras/LengthFieldPrepender.swift

```diff
@@ -13,6 +13,27 @@
 //===----------------------------------------------------------------------===//
 
 import NIO
+import Foundation
+
+extension ByteBuffer {
+    @discardableResult
+    @inlinable
+    mutating func write24UInt(
+        _ integer: UInt32,
+        endianness: Endianness = .big
+    ) -> Int {
+        assert(integer & 0xFF_FF_FF == integer, "integer value does not fit into 24 bit integer")
```

Comment:
I'd make this a `precondition` so it's also checked in release mode.

Reply:
Done